File: program/program/find-average-of-numbers-by-recursion/find-average-of-numbers-by-recursion.jl
function average_recursive(numbers, n)
    if n == 0
        return 0.0
    else
        return (numbers[n] + (n - 1) * average_recursive(numbers, n - 1)) / n
    end
end

function main()
    println("Enter numbers separated by spaces:")
    input = readline()
    numbers = [parse(Float64, num) for num in split(input)]
    avg = average_recursive(numbers, length(numbers))
    println("Average:", avg)
end

main()

File: program/program/find-average-of-numbers-by-recursion/find_average_of_numbers_by_recursion.rs
use std::io;
use std::str::FromStr;

fn average_recursive(numbers: &[i32], sum: f64, count: usize) -> f64 {
    match numbers {
        [] => sum / count as f64,
        [first, rest @ ..] => {
            let new_sum = sum + *first as f64;
            let new_count = count + 1;
            average_recursive(rest, new_sum, new_count)
        }
    }
}

fn main() {
    println!("Enter numbers separated by space:");

    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let numbers: Vec<i32> = input
        .split_whitespace()
        .filter_map(|num| i32::from_str(num).ok())
        .collect();

    if numbers.is_empty() {
        println!("No numbers entered.");
        return;
    }

    let avg = average_recursive(&numbers, 0.0, 0);
    println!("The average is {}", avg);
}
